bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -584,7 +586,20 @@ public static boolean checkIsType(BType sourceType, BType targetType, List<TypeP
                 if (sourceType.getTag() == TypeTags.FINITE_TYPE_TAG) {
                     return isFiniteTypeMatch((BFiniteType) sourceType, targetType);
                 }
-                return TypeTags.isXMLTypeTag(sourceType.getTag());
+                if (sourceType.getTag() == TypeTags.XML_TAG) {
+                    BXMLType target = ((BXMLType) targetType);
+                    BType targetConstraint = target.constraint;
+                    // TODO: Revisit and check why xml<xml<constraint>>> on chained iteration
+                    while (target.constraint.getTag() == TypeTags.XML_TAG) {
+                        target = (BXMLType) target.constraint;
+                        targetConstraint = target.constraint;
+                    }
+                    return checkIsType(((BXMLType) sourceType).constraint, targetConstraint,
+                            unresolvedTypes);
+                } else if (TypeTags.isXMLTypeTag(sourceType.getTag())) {
+                    return true;
```

Comment:
Doesn't the chained constraints (like in the above case) apply for subtypes?

Reply:
The constraints only apply for XML types.